### PR TITLE
feat: basic support for 8x8 jaas

### PR DESCRIPTION
**Pull request by debdutdeb**
Only gurantees support for 8x8 jaas, with no toggleable features for now.

### CODE REVIEW

**Comment by debdutdeb** [comment]
// TODO: update app version

**Comment by debdutdeb** [comment]
@pierre-lehnen-rc fyi

File: src/videoConfProvider.ts [hunk 1]
```diff
@@ -129,19 +135,30 @@ export class JitsiProvider implements IVideoConfProvider {
...
+			Object.assign(header, { kid: this.jaasApiKey, alg: 'RS256' });
+
+			payload.iss = 'chat';
+			payload.sub = this.jitsiAppId;
+			payload.aud = 'jitsi';
+			payload.context.features = {};
+
+			return jws.JWS.sign('RS256', JSON.stringify(header), JSON.stringify(payload), new Buffer(this.jaasPrivateKey, 'base64').toString('utf8'));
+		}
+
```
Comment (pierre-lehnen-rc) [inline src/videoConfProvider.ts:161]:
```suggestion
		if (this.useJaaS) {
			const jaasHeader = {
				...header,
				{
					kid: this.jaasApiKey,
					alg: 'RS256',
				},
			};

			const jaasPayload = {
				...payload,
				iss: 'chat',
				sub: this.jitsiAppId,
				aud: 'jitsi',
				context: {
					...(payload.context || {}),
					features: {},
				},
			}

			return jws.JWS.sign('RS256', JSON.stringify(jaasHeader), JSON.stringify(jaasPayload), new Buffer(this.jaasPrivateKey, 'base64').toString('utf8'));
		}

```

File: src/JitsiApp.ts [hunk 2]
```diff
@@ -40,10 +41,22 @@ export class JitsiApp extends App {
...
+		provider.jaasApiKey = await settings.getValueById(AppSetting.JitsiApplicationId);
 
 		return true;
 	}
 
+	async onPreSettingUpdate(context: ISettingUpdateContext, configurationModify: IConfigurationModify, read: IRead, http: IHttp): Promise<ISetting> {
+		if (context.newSetting.id === AppSetting.JitsiTokenAuditor) {
+			if (this.getProvider()?.useJaaS) {
+				context.newSetting.value = 'jitsi';
+			}
```
Comment (pierre-lehnen-rc) [inline src/JitsiApp.ts:55]: What's this for?

**Review by pierre-lehnen-rc — COMMENTED** [review]
Optional changes (except the isFullyConfigured stuff)

File: src/JitsiApp.ts (hunk 2 above)
Comment (debdutdeb) [reply src/JitsiApp.ts:55]: 8x8 requires the audience to always be jitsi, figured we then make sure it is reflected in the setting always.

Comment (pierre-lehnen-rc) [reply src/JitsiApp.ts:55]: Does the UI handle this properly? As in showing the new value on the setting field automatically when the settings are saved.

File: app.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "3b387ba9-f57c-44c6-9810-8c0256abd64c",
-    "version": "2.0.4",
+    "version": "2.1.4",
```
Comment (pierre-lehnen-rc) [inline app.json:3]:
```suggestion
    "version": "2.1.0",
```

File: src/JitsiApp.ts (hunk 2 above)
Comment (debdutdeb) [reply src/JitsiApp.ts:55]: probably on a rerender, not immediately. I did not check

Comment (pierre-lehnen-rc) [reply src/JitsiApp.ts:55]: Tried it here and this "jitsi" value only gets applied if the user tries to set something else to the auditor setting, it doesn't get applied automatically when enabling JaaS which I assume was the intended behavior.

I believe it's best to take this out and just document this requirement.

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]

File: src/JitsiApp.ts [hunk 4]
```diff
@@ -40,10 +41,22 @@ export class JitsiApp extends App {
...
+		}
+		return context.newSetting;
+	}
+
```
Comment (pierre-lehnen-rc) [inline src/JitsiApp.ts:59]:
```suggestion
```

**Review by pierre-lehnen-rc — APPROVED** [review]

**Review by AllanPazRibeiro — APPROVED** [review]

**Pull request merged**